test/internals/reply.test.js

```diff
@@ -1640,6 +1654,67 @@ test('reply.getResponseTime() should return the same value after a request is fi
   fastify.inject({ method: 'GET', url: '/' })
 })
 
+test('reply.elapsedTime should return a number greater than 0 after the timer is initialised on the reply by setting up response listeners', t => {
+  t.plan(1)
+  const fastify = Fastify()
+  fastify.route({
+    method: 'GET',
+    url: '/',
+    handler: (req, reply) => {
+      reply.send('hello world')
+    }
+  })
+
+  fastify.addHook('onResponse', (req, reply) => {
+    t.ok(reply.elapsedTime > 0)
+    t.end()
+  })
+
+  fastify.inject({ method: 'GET', url: '/' })
+})
+
+test('reply.elapsedTime should return the time since a request started while inflight', t => {
+  t.plan(1)
+  const fastify = Fastify()
+  fastify.route({
+    method: 'GET',
+    url: '/',
+    handler: (req, reply) => {
+      reply.send('hello world')
+    }
+  })
+
+  fastify.addHook('preValidation', (req, reply, done) => {
+    t.not(reply.elapsedTime, reply.elapsedTime)
+    done()
+  })
+
+  fastify.addHook('onResponse', (req, reply) => {
+    t.end()
+  })
```

Comment:
Maybe refactor this test to compare the `elapsedTime` within `preValidation` and later within `onResponse` to verify the latter is greater than the former?